### PR TITLE
[DRAFT] Add version description component

### LINKED ISSUE #454: Create "What's changing" UI element
https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/420954309

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made?

Fixes #454 

Screen Shot 2021-01-29 at 9 07 15 AM

Screen Shot 2021-01-29 at 9 07 09 AM

## How was this change tested?

IN PROGRESS

### CODE REVIEW

File: app/javascript/stylesheets/work.scss [hunk 1]
```diff
@@ -58,3 +58,11 @@
     margin-left: 10px;
   }
 }
+
+.work-editor {
+  .version {
+    background-color: $stanford-digital-pink;
+    border-color: $stanford-cardinal;
+    border-width: 3px;
+  }
```
Comment (mjgiarlo) [inline app/javascript/stylesheets/work.scss:67]: I believe you should be able to move these lines into `app/javascript/stylesheets/workEditor.scss` which already has a `.work-editor` block in it.

File: app/javascript/stylesheets/palette.scss [hunk 2]
```diff
@@ -1,9 +1,12 @@
 $alabaster: #f2f1eb;
 $silver-sand: #c0c0bf; // "30% black" in Stanford brand identity
 $stanford-cardinal: #8c1515;
+$stanford-cardinal-light: #B83A4B;
+$stanford-spirited-light: #F4795B;
```
Comment (mjgiarlo) [inline app/javascript/stylesheets/palette.scss:5]: Cruft now?

File: app/javascript/stylesheets/palette.scss [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 $stanford-cool-grey: #53565a; // used in SDR header (not yet added)
 $stanford-digital-blue: #006cb8;
 $stanford-digital-green: #008566;
 $stanford-digital-red: #b1040e;
+$stanford-digital-pink: #F3E7E7;
```
Comment (mjgiarlo) [inline app/javascript/stylesheets/palette.scss:10]: This is TOTALLY a nitpick, but the other colors in `palette.scss` use lowercase hex digits and the new entry uses uppercase.

File: app/controllers/works_controller.rb [hunk 4]
```diff
@@ -92,6 +92,8 @@ def work_form(work)
   sig { params(work: Work).void }
   def after_save(work)
     work.event_context = { user: current_user }
+    work.event_context[:description] = params[:work][:version_description] unless params[:work][:version_description].nil?
```
Comment (mjgiarlo) [inline app/controllers/works_controller.rb:95]: Would you consider:

```suggestion
work.event_context = { user: current_user, description: params.dig(:work, :version_description) }.compact
```

:question:

File: app/components/works/version_component.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+# frozen_string_literal: true
+
+module Works
+  # Renders a widget for setting the title and contact for the work.
+  class VersionComponent < ApplicationComponent
+    def initialize(form:)
+      @form = form
+    end
+
+    attr_reader :form
```
Comment (mjgiarlo) [inline app/components/works/version_component.rb:11]: Should the component render no matter what state the work is in? Or should it only render in certain states? Might be a PO question. cc: @amyehodge @jcoyne

**Review by mjgiarlo — COMMENTED** [review]
Looking good so far! @amyehodge do we want this for collections as well as works?

File: app/controllers/works_controller.rb (hunk 4 above)
Comment (aaron-collier) [reply app/controllers/works_controller.rb:95]: I like.

**Comment by amyehodge** [comment]
@mjgiarlo , that makes a lot of sense to me. It can provide useful information and frankly is a good reminder that you're versioning the collection. It is not in the spec at all. Should I make a new ticket for that?

File: app/components/works/version_component.rb (hunk 5 above)
Comment (aaron-collier) [reply app/components/works/version_component.rb:11]: That's a good question. I figured as this is the edit form it probably always needs to be there... but validation/etc?

**Comment by mjgiarlo** [comment]
@amyehodge :speech_balloon: 
> @mjgiarlo , that makes a lot of sense to me. It can provide useful information and frankly is a good reminder that you're versioning the collection. It is not in the spec at all. Should I make a new ticket for that?

I'd defer to @aaron-collier on that.

File: app/javascript/stylesheets/palette.scss (hunk 3 above)
Comment (mjgiarlo) [reply app/javascript/stylesheets/palette.scss:10]: I wonder if we want a different name, too, so as not to suggest that this is an official Stanford identity color. When googling that hex code, I see that some paint companies sell this as `Laughing Maiden` or `Ibis White`.

How about `$ibis-white: #f3e7e7; // light pinkish hue used in what's changing box` here?

File: app/components/works/version_component.rb (hunk 5 above)
Comment (amyehodge) [reply app/components/works/version_component.rb:11]: It needs to be there for VERSION drafts, but not for the initial drafts. In other words, it should always be there for drafts of previously accessioned content. 

A user may open this draft and save it multiple times, so 1) any entry should be saved when the draft is saved and continue to be editable until the deposit is made and 2) be required to make the deposit.

**Pull request closed without merging**